src/main/java/fittrack/parser/Parser.java

```diff
@@ -0,0 +1,280 @@
+            printAddedReminder(reminderList);
+            break;
+        case DELETE_REMINDER_COMMAND:
+            int reminderIndexToDelete = Integer.parseInt(description) - 1;
+            assert reminderIndexToDelete >= 0 && reminderIndexToDelete < reminderList.size() : "Delete reminder index "
+                    + "out of bounds";
+            Reminder reminderToDelete = reminderList.get(reminderIndexToDelete);
+            reminderList.remove(reminderIndexToDelete);
+            printDeletedReminder(reminderList, reminderToDelete);
+            break;
+        case LIST_REMINDER_COMMAND:
+            printReminderList(reminderList);
+            break;
+        case LIST_UPCOMING_REMINDER_COMMAND:
+            beginSegment();
+            printUpcomingReminders(reminderList);
+            break;
+
+        case "add-goal":  // use "add-goal" consistently in input and command handling
+            if (!description.isEmpty()) {
+                String[] goalParts = description.split(" ", 2);
+                String goalDescription = goalParts[0];
+                LocalDateTime goalDeadline = null;
+
+                if (goalParts.length > 1) {
+                    String goalDeadlineInput = goalParts[1];
+                    try {
+                        goalDeadline = parseGoalDeadline(goalDeadlineInput);
+                    } catch (IllegalArgumentException e) {
+                        System.out.println("Invalid date format: " + e.getMessage());
+                        return;
+                    }
+                }
+                Goal newGoal = new Goal(goalDescription, goalDeadline);
+                goalList.add(newGoal);
+                printAddedGoal(goalList);
+            } else {
+                System.out.println("Please specify a goal to add.");
+            }
+            break;
+
+        case "delete-goal":
+            try {
+                int index = Integer.parseInt(description) - 1;
+                if (index >= 0 && index < goalList.size()) {
+                    goalList.remove(index);
+                    System.out.println("Goal at index " + (index + 1) + " has been removed.");
+                } else {
+                    System.out.println("Invalid goal index.");
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Please specify a valid index to delete.");
+            }
+            break;
+
+        case "list-goal":
+            if (goalList.isEmpty()) {
+                System.out.println("No goals to display.");
+            } else {
+                System.out.println("Goals:");
+                for (int i = 0; i < goalList.size(); i++) {
+                    System.out.println((i + 1) + ". " + goalList.get(i));
+                }
+            }
+            break;
+
+        default:
+            printUnrecognizedInputMessage(); // Response to unrecognized inputs
+            break;
+        }
+    }
```

Comment:
Method is too long. Do consider extracting out code for each command to a separate method